crates/ruff_server/src/message.rs

```diff
@@ -10,53 +10,31 @@ pub(crate) fn init_messenger(client_sender: ClientSender) {
     MESSENGER
         .set(client_sender)
         .expect("messenger should only be initialized once");
-
-    // unregister any previously registered panic hook
-    let _ = std::panic::take_hook();
-
-    // When we panic, try to notify the client.
-    std::panic::set_hook(Box::new(move |panic_info| {
-        if let Some(messenger) = MESSENGER.get() {
-            let _ = messenger.send(lsp_server::Message::Notification(
-                lsp_server::Notification {
-                    method: lsp_types::notification::ShowMessage::METHOD.into(),
-                    params: serde_json::to_value(lsp_types::ShowMessageParams {
-                        typ: lsp_types::MessageType::ERROR,
-                        message: String::from(
-                            "The Ruff language server exited with a panic. See the logs for more details."
-                        ),
-                    })
-                    .unwrap_or_default(),
-                },
-            ));
-        }
-
-        let backtrace = std::backtrace::Backtrace::force_capture();
-        tracing::error!("{panic_info}\n{backtrace}");
-        #[allow(clippy::print_stderr)]
-        {
-            // we also need to print to stderr directly in case tracing hasn't
-            // been initialized.
-            eprintln!("{panic_info}\n{backtrace}");
-        }
-    }));
```

Comment:
I decided to move this out of Messenger to ensure that we unset the panic hook when the server shuts down. We may consider using `catch_unwind` on a per request/response model in the future. But I think that's good enough for now